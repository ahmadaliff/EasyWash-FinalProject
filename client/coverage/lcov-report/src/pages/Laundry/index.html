
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/pages/Laundry</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/pages/Laundry</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">59.25% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>32/54</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">35% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>7/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">55.55% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/18</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">61.22% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>30/49</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="action.js"><a href="action.js.html">action.js</a></td>
	<td data-value="66.66" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="constants.js"><a href="constants.js.html">constants.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="index.jsx"><a href="index.jsx.html">index.jsx</a></td>
	<td data-value="90" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="20" class="abs high">18/20</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	</tr>

<tr>
	<td class="file medium" data-value="reducer.js"><a href="reducer.js.html">reducer.js</a></td>
	<td data-value="71.42" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 71%"></div><div class="cover-empty" style="width: 29%"></div></div>
	</td>
	<td data-value="71.42" class="pct medium">71.42%</td>
	<td data-value="7" class="abs medium">5/7</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="71.42" class="pct medium">71.42%</td>
	<td data-value="7" class="abs medium">5/7</td>
	</tr>

<tr>
	<td class="file low" data-value="saga.js"><a href="saga.js.html">saga.js</a></td>
	<td data-value="11.11" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 11%"></div><div class="cover-empty" style="width: 89%"></div></div>
	</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="18" class="abs low">2/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="16.66" class="pct low">16.66%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="18" class="abs low">2/18</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-12-29T03:49:59.223Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    