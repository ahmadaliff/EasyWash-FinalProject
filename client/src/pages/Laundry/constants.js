export const GET_MERCHANT = 'Laundry/GET_MERCHANT';
export const SET_MERCHANT = 'Laundry/SET_MERCHANT';
export const RESET_MERCHANT = 'Laundry/RESET_MERCHANT';
