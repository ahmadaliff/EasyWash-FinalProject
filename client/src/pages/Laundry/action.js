import { GET_MERCHANT, RESET_MERCHANT, SET_MERCHANT } from './constants';

export const actionGetMerchant = (id) => ({
  type: GET_MERCHANT,
  id,
});

export const actionSetMerchant = (data) => ({
  type: SET_MERCHANT,
  data,
});

export const actionResetMerchant = () => ({
  type: RESET_MERCHANT,
});
