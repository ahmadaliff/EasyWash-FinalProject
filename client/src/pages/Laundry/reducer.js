import { produce } from 'immer';
import { RESET_MERCHANT, SET_MERCHANT } from './constants';

export const initialState = {
  merchant: null,
};

export const merchantReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_MERCHANT:
        draft.merchant = action.data;
        break;
      case RESET_MERCHANT:
        return initialState;
    }
  });
