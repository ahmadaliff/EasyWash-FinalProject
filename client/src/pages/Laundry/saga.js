import { call, put, takeLatest } from 'redux-saga/effects';
import toast from 'react-hot-toast';

import intlHelper from '@utils/intlHelper';

import { apiGetMerchantById } from '@domain/api';

import { GET_MERCHANT } from '@pages/Laundry/constants';
import { actionSetMerchant } from '@pages/Laundry/action';
import { setLoading, showPopup } from '@containers/App/actions';

const getUserLocation = () =>
  new Promise((resolve, reject) => {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(
        (location) => {
          const { latitude, longitude } = location.coords;
          const loc = { location: JSON.stringify({ lat: latitude, lng: longitude }) };
          resolve(loc);
        },
        (error) => reject(error)
      );
    }
  });

function* sagaGetMerchant({ id }) {
  yield put(setLoading(true));
  try {
    const location = yield call(getUserLocation);
    const response = yield call(apiGetMerchantById, id, location);
    yield put(actionSetMerchant(response.data));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404 || error?.response?.status === 401) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

export default function* merchantSaga() {
  yield takeLatest(GET_MERCHANT, sagaGetMerchant);
}
