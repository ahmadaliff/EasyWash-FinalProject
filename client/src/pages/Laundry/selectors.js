import { createSelector } from 'reselect';
import { initialState } from '@pages/Laundry/reducer';

const selectMerchantState = (state) => state.merchant || initialState;

export const selectMerchant = createSelector(selectMerchantState, (state) => state.merchant);
