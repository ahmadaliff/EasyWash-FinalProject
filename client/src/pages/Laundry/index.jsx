import PropTypes from 'prop-types';
import { useEffect, useState } from 'react';
import { connect, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { useNavigate, useParams } from 'react-router-dom';
import { FormattedMessage, FormattedNumber } from 'react-intl';

import { Button, Card, CardContent } from '@mui/material';
import { ArrowBack, DryCleaning } from '@mui/icons-material';

import NoData from '@components/NoData';
import MerchantInfo from '@components/MerchantInfo';
import { selectMerchant } from '@pages/Laundry/selectors';

import ServiceDialog from '@components/ServiceDialog';

import { actionGetMerchant, actionResetMerchant } from '@pages/Laundry/action';

import classes from '@pages/Laundry/style.module.scss';

const Laundry = ({ merchant }) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { id } = useParams();
  const [openDialog, setOpenDialog] = useState(false);
  const [serviceProps, setServiceProps] = useState(null);

  useEffect(() => {
    if (!merchant) dispatch(actionGetMerchant(id));
    return () => {
      if (merchant) dispatch(actionResetMerchant());
    };
  }, [dispatch, id, merchant]);

  return (
    <main className={classes.mainWrap}>
      <div className={classes.header}>
        <h3>
          <FormattedMessage id="app_merchant_header" />
        </h3>
        <Button
          variant="contained"
          className={classes.backButton}
          startIcon={<ArrowBack />}
          onClick={() => navigate(-1)}
        >
          <FormattedMessage id="app_back" />
        </Button>
      </div>
      <Card className={classes.card}>
        <CardContent className={classes.cardContent}>
          <MerchantInfo merchant={merchant} />
          {merchant?.Services?.map((service, k) => (
            <div key={k}>
              <div
                className={classes.service}
                onClick={() => {
                  setOpenDialog(true);
                  setServiceProps(service);
                }}
              >
                <DryCleaning />
                <div className={classes.serviceInfoWrap}>
                  <div className={classes.serviceInfo}>
                    <p>{service?.name}</p>
                    <p>
                      <FormattedNumber
                        value={service?.price}
                        // eslint-disable-next-line react/style-prop-object
                        style="currency"
                        currency="IDR"
                        minimumFractionDigits={0}
                      />
                    </p>
                  </div>
                </div>
              </div>
            </div>
          ))}
          <ServiceDialog handleClose={() => setOpenDialog(false)} open={openDialog} service={serviceProps} />
          {merchant?.Services?.length === 0 && <NoData />}
          {!merchant && <NoData />}
        </CardContent>
      </Card>
    </main>
  );
};

Laundry.propTypes = {
  merchant: PropTypes.object,
};
const mapStateToProps = createStructuredSelector({
  merchant: selectMerchant,
});
export default connect(mapStateToProps)(Laundry);
