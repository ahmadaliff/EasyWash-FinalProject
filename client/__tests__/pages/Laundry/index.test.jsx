import Laundry from '@pages/Laundry';
import { render, fireEvent, act } from '@utils/testHelper';

let wrapper;
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

jest.mock('@pages/Laundry/selectors', () => ({
  selectMerchant: () => ({
    imagePath: null,
    name: 'laundry',
    description: 'description',
    Services: [
      {
        name: 'reguler',
        isUnit: false,
        price: 2000,
      },
    ],
  }),
}));

beforeEach(() => {
  act(() => {
    wrapper = render(<Laundry />);
  });
});

describe('Laundry Page', () => {
  test('render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('laundry-page')).toBeInTheDocument();
  });
  test('navigate back when back button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('back-button');
    await act(async () => {
      fireEvent.click(button);
    });
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
  test('open dialog when button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-dialog');
    act(() => {
      fireEvent.click(button);
    });
    expect(getByTestId('dialog-service')).toBeInTheDocument();
  });
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
