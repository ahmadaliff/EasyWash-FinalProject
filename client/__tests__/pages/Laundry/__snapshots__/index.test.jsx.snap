// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Laundry Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <main
        class="mainWrap"
        data-testid="laundry-page"
      >
        <div
          class="header"
        >
          <h3>
            Laundry
          </h3>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="back-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root cardContent css-46bh2p-MuiCardContent-root"
          >
            <div
              class="merchant"
              data-testid="merchant-info"
            >
              <div
                class="imgWrap"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
                >
                  l
                </div>
              </div>
              <div>
                <h4
                  class="merchantInfoHeader"
                >
                  laundry
                </h4>
                <span
                  class="profileInfo"
                >
                  <p
                    class="profileText"
                  >
                    description
                  </p>
                </span>
              </div>
              <button
                aria-label="Pesan"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium chatButton css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AddCommentOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4zm-2 13.17L18.83 16H4V4h16v13.17zM13 5h-2v4H7v2h4v4h2v-4h4V9h-4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div>
              <div
                class="service"
                data-testid="button-dialog"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DryCleaningIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.56 11.36 13 8.44V7c0-.55-.45-1-1-1s-1-.45-1-1 .45-1 1-1 1 .45 1 1h2c0-1.84-1.66-3.3-3.56-2.95-1.18.22-2.15 1.17-2.38 2.35-.3 1.56.6 2.94 1.94 3.42v.63l-6.56 2.92c-.88.38-1.44 1.25-1.44 2.2v.01C3 14.92 4.08 16 5.42 16H7v6h10v-6h1.58c1.34 0 2.42-1.08 2.42-2.42v-.01c0-.95-.56-1.82-1.44-2.21zM18.58 14H17v-1H7v1H5.42c-.23 0-.42-.19-.42-.43 0-.17.1-.32.25-.38l6.75-3 6.75 3c.15.07.25.22.25.39 0 .23-.19.42-.42.42z"
                  />
                </svg>
                <div
                  class="serviceInfoWrap"
                >
                  <div
                    class="serviceInfo"
                  >
                    <div
                      class="Name"
                    >
                      <p>
                        reguler
                      </p>
                      <p
                        class="type"
                      >
                        Tipe  
                        Satuan
                      </p>
                    </div>
                    <p>
                      RpÂ 2.000
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="PrivateSwipeArea-root css-17j2vre"
              style="height: 20px;"
            />
          </div>
        </div>
      </main>
    </div>
    <div
      aria-hidden="true"
      class="MuiDrawer-root MuiDrawer-modal dialog MuiModal-root MuiModal-hidden css-12y4yjv-MuiModal-root-MuiDrawer-root"
      data-testid="dialog-service"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorBottom css-9emuhu-MuiPaper-root-MuiDrawer-paper"
        style="visibility: hidden; pointer-events: none; webkit-transform: translateY(768px); transform: translateY(768px);"
        tabindex="-1"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1xi8exm-MuiTypography-root-MuiDialogTitle-root"
        >
          <b>
            Detail Layanan
          </b>
        </h2>
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-qsfli3-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div>
            <div
              class="field"
            >
              <p
                class="label"
              >
                Nama Layanan
              </p>
              <p
                class="text"
              />
            </div>
            <div
              class="field"
            >
              <p
                class="label"
              >
                Harga
              </p>
              <p
                class="text"
              >
                RpNaN
              </p>
            </div>
            <div
              class="field"
            >
              <p
                class="label"
              >
                Tipe  
              </p>
              <p
                class="text"
              >
                Satuan
              </p>
            </div>
          </div>
          <div
            class="buttonQuantity"
          >
            <button
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="button-min"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="RemoveIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13H5v-2h14v2z"
                />
              </svg>
            </button>
            1
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="button-plus"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing action css-knqc4i-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium dialogAction css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-add"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
            </span>
            Tambahkan ke keranjang
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium dialogAction css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-create"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LibraryBooksOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H8V4h12v12zM10 9h8v2h-8zm0 3h4v2h-4zm0-6h8v2h-8z"
                />
              </svg>
            </span>
            Buat Pesanan
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="laundry-page"
    >
      <div
        class="header"
      >
        <h3>
          Laundry
        </h3>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="back-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Kembali
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root cardContent css-46bh2p-MuiCardContent-root"
        >
          <div
            class="merchant"
            data-testid="merchant-info"
          >
            <div
              class="imgWrap"
            >
              <div
                class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
              >
                l
              </div>
            </div>
            <div>
              <h4
                class="merchantInfoHeader"
              >
                laundry
              </h4>
              <span
                class="profileInfo"
              >
                <p
                  class="profileText"
                >
                  description
                </p>
              </span>
            </div>
            <button
              aria-label="Pesan"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium chatButton css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddCommentOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4zm-2 13.17L18.83 16H4V4h16v13.17zM13 5h-2v4H7v2h4v4h2v-4h4V9h-4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div>
            <div
              class="service"
              data-testid="button-dialog"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DryCleaningIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.56 11.36 13 8.44V7c0-.55-.45-1-1-1s-1-.45-1-1 .45-1 1-1 1 .45 1 1h2c0-1.84-1.66-3.3-3.56-2.95-1.18.22-2.15 1.17-2.38 2.35-.3 1.56.6 2.94 1.94 3.42v.63l-6.56 2.92c-.88.38-1.44 1.25-1.44 2.2v.01C3 14.92 4.08 16 5.42 16H7v6h10v-6h1.58c1.34 0 2.42-1.08 2.42-2.42v-.01c0-.95-.56-1.82-1.44-2.21zM18.58 14H17v-1H7v1H5.42c-.23 0-.42-.19-.42-.43 0-.17.1-.32.25-.38l6.75-3 6.75 3c.15.07.25.22.25.39 0 .23-.19.42-.42.42z"
                />
              </svg>
              <div
                class="serviceInfoWrap"
              >
                <div
                  class="serviceInfo"
                >
                  <div
                    class="Name"
                  >
                    <p>
                      reguler
                    </p>
                    <p
                      class="type"
                    >
                      Tipe  
                      Satuan
                    </p>
                  </div>
                  <p>
                    RpÂ 2.000
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="PrivateSwipeArea-root css-17j2vre"
            style="height: 20px;"
          />
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
